Sources/MongoSwift/ConnectionString.swift

```diff
@@ -5,45 +5,33 @@ internal class ConnectionString {
     /// Pointer to the underlying `mongoc_uri_t`.
     internal let _uri: OpaquePointer
 
-    /// Initializes a new `ConnectionString` with the provided options. Updates `options` to correctly reflect the
-    /// final options set on the `ConnectionString`.
-    internal init(_ connectionString: String, options: inout ClientOptions) throws {
+    /// Initializes a new `ConnectionString` with the provided options.
+    internal init(_ connectionString: String, options: ClientOptions? = nil) throws {
         var error = bson_error_t()
         guard let uri = mongoc_uri_new_with_error(connectionString, &error) else {
             throw extractMongoError(error: error)
         }
         self._uri = uri
 
-        if let rc = options.readConcern {
+        if let rc = options?.readConcern {
             self.readConcern = rc
         }
-        options.readConcern = self.readConcern
 
-        if let wc = options.writeConcern {
+        if let wc = options?.writeConcern {
             self.writeConcern = wc
         }
-        options.writeConcern = self.writeConcern
 
-        if let rp = options.readPreference {
+        if let rp = options?.readPreference {
             self.readPreference = rp
         }
-        options.readPreference = self.readPreference
 
-        if let rw = options.retryWrites {
+        if let rw = options?.retryWrites {
             mongoc_uri_set_option_as_bool(self._uri, MONGOC_URI_RETRYWRITES, rw)
         }
-        if let rr = options.retryReads {
-            mongoc_uri_set_option_as_bool(self._uri, MONGOC_URI_RETRYREADS, rr)
-        }
 
-        // we can't get values for retryReads and retryWrites individually, so we will read
```

Comment:
the client doesn't get the options struct back anymore, and doesn't actually care what the values of retryReads and retryWrites are, so we don't need to worry about retrieving those values anymore to send back.